File: src/main/java/com/mbayou/kick4k/events/type/EventUser.java
package com.mbayou.kick4k.events.type;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class EventUser {
    private final Boolean isAnonymous;
    private final Integer userId;
    private final String username;
    private final Boolean isVerified;
    private final String profilePicture;
    private final String channelSlug;
    private final Identity identity;

    @JsonCreator
    public EventUser(@JsonProperty("is_anonymous") Boolean isAnonymous,
                     @JsonProperty("user_id") Integer userId,
                     @JsonProperty("username") String username,
                     @JsonProperty("is_verified") Boolean isVerified,
                     @JsonProperty("profile_picture") String profilePicture,
                     @JsonProperty("channel_slug") String channelSlug,
                     @JsonProperty("identity") Identity identity) {
        this.isAnonymous = isAnonymous;
        this.userId = userId;
        this.username = username;
        this.isVerified = isVerified;
        this.profilePicture = profilePicture;
        this.channelSlug = channelSlug;
        this.identity = identity;
    }

    public Boolean getAnonymous() {
        return isAnonymous;
    }

    public Integer getUserId() {
        return userId;
    }

    public String getUsername() {
        return username;
    }

    public Boolean getVerified() {
        return isVerified;
    }

    public String getProfilePicture() {
        return profilePicture;
    }

    public String getChannelSlug() {
        return channelSlug;
    }

    public Identity getIdentity() {
        return identity;
    }
}

File: src/main/kotlin/com/mbayou/kick4k/events/type/EventUser.kt
package com.mbayou.kick4k.events.type

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

data class EventUser @JsonCreator constructor(
    @JsonProperty("is_anonymous") val isAnonymous: Boolean?,
    @JsonProperty("user_id") val userId: Int?,
    @JsonProperty("username") val username: String?,
    @JsonProperty("is_verified") val isVerified: Boolean?,
    @JsonProperty("profile_picture") val profilePicture: String?,
    @JsonProperty("channel_slug") val channelSlug: String?,
    @JsonProperty("identity") val identity: Identity?,
)
